Added UI for taxonomy terms

harumijang opened the pull request.

harumijang commented:
Added interfaces for `taxonomy--person_type` so far.
Can test by hitting `taxonomy/term/3`, `taxonomy/term/4,` `taxonomy/term/6` if you have the acms demo content.

Will continue adding the rest of the taxonomy vocabulary tomorrow. 

![image](https://user-images.githubusercontent.com/35243020/177659257-ac0b995e-5dd4-400f-bb6c-24dea2423dd3.png)

![image](https://user-images.githubusercontent.com/35243020/177659277-53340f82-d8cb-4ac4-a764-afb7ad2303e5.png)

harumijang commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -21,16 +25,57 @@ const CONTENT_TYPES = [
-  return (
+  return Array.isArray(node) ? (
+    <Layout title={label} menus={menus}>
+      {type === 'taxonomy_term--article_type' &&
```
I am checking the `type` so existing taxonomy terms that don't have any nodes will still render a page with the term shown on the page

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -21,16 +25,57 @@ const CONTENT_TYPES = [
+  'taxonomy_term--place_type',
 ];
 
 interface NodePageProps extends LayoutProps {
```
Hmm, `NodePageProps` sounds specific to nodes. I think we need some refactoring to make this generic so that it makes sense for both, taxonomy terms and nodes.

lauriii replied:
Maybe we should convert some of the logic into components and then this would become "EntityPage" or something along the lines of that

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -112,9 +163,81 @@ export async function getStaticProps(
-  });
+  let node;
+  if (type === 'taxonomy_term--person_type') {
+    node = await drupal.getResourceCollectionFromContext<DrupalNode[]>(
```
I think it's confusing that we use the `node` property both for a list of nodes and for a single node. I think we should change this something else and make it so that both taxonomy term and list of nodes get passed to the component.

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -77,6 +122,12 @@ export async function getStaticProps(
+  let label;
+  if (path.label) {
+    // Get the taxonomy term from the path.
+    label = path.label;
```
For consistency, we should probably get the title for taxonomy terms from the entity object like we do for nodes

harumijang replied:
What if there is an existing taxonomy term with no associated objects like 'Software' for example

![image](https://user-images.githubusercontent.com/35243020/179066527-271bfb56-45ff-4f77-b96c-f548ece40b6a.png)

lauriii replied:
The taxonomy term object should be still available, even if there aren't any associated entities.

harumijang replied:
Ah yeah, I misunderstood your first comment.

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -21,24 +22,43 @@ const CONTENT_TYPES = [
-
-  return (
+export default function EntityPage({
+  node,
```
Can we call this entity which type would be either `DrupalNode` or `DrupalTaxonomyTerm`?

harumijang replied:
Could you rephrase this? I don't understand

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -21,24 +22,43 @@ const CONTENT_TYPES = [
-  return (
+export default function EntityPage({
+  node,
+  node_list,
```
I'd call this something more generic. Maybe `additionalContent` could work?

lauriii commented on starters/basic-starter/components/taxonomy/taxonomy-term.tsx:
```diff
@@ -0,0 +1,43 @@
+  return (
+    <Layout title={taxonomy_term} menus={menus}>
+      {type === 'taxonomy_term--article_type' &&
+        node_list.every(
```
I think we could actually potentially remove this check and move this handling to the individual term components. This would probably simplify this enough so that we could remove this component completely and move it back to the slug.

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -111,32 +137,130 @@ export async function getStaticProps(
+        context,
+        {
+          params: new DrupalJsonApiParams()
+            .addInclude([
```
We should probably sort all of the lists. Maybe sorting by the creation time would make sense (this would be same as what we do in Drupal)?

harumijang commented:
Addressed feedback, just need clarification on this comment:

> Can we call this entity which type would be either `DrupalNode` or `DrupalTaxonomyTerm`?

harumijang commented:
Confirmed the taxonomy term links used in the test with a new fresh install with demo content

lauriii commented on starters/basic-starter/pages/[...slug].tsx:
```diff
@@ -13,30 +13,84 @@ import { NodePerson } from 'components/node--person';
+        />
+      )}
+      {entity.type === 'taxonomy_term--categories' && (
+        <TaxonomyArticle
```
Should we explain with an inline comment, why are we rendering the categories with TaxonomyArticle? I'd be fine with removing support for categories since it makes the least sense without having support for including multiple content types

harumijang replied:
I am fine with removing it too since it's a demo
